scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -612,13 +612,14 @@ class Router(formatOps: FormatOps) {
           .getOrElse(getLastToken(leftOwner))
         val forceNewlineBeforeExtends = Policy.before(expire) {
           case Decision(t @ FormatToken(_, soft.ExtendsOrDerives(), _), s)
-              if t.meta.rightOwner == leftOwner =>
+              if t.meta.rightOwner.parent.contains(leftOwner) =>
```

Comment:
this change is because `extends` is now part of template, not `object/class`.

this also means that this never matched `derives`.